### ISSUE TITLE
AddHostObjectToScript possible bug

### ISSUE THREAD

**Issue by GrandStrateguerre** [MicrosoftEdge/WebView2Feedback#1628]
Hi,
I can not get the AddHostObjectToScript works.

Here is the code I've used :
Class to be Hosted: 
```cs
 // Bridge  
        [ClassInterface(ClassInterfaceType.AutoDual)]
        [ComVisible(true)]
        public class bridge 
        {
            [ComVisible(true)]
            public string Test(string param)
            {
                return "Test : " + param;
            }

        }
```

Lines of code in C# side to add the hosted object in webview2 :
```cs
    private void WebView_CoreWebView2InitializationCompleted(object sender, CoreWebView2InitializationCompletedEventArgs e)
        {
           .....
            //
            try
            {
                webView21.CoreWebView2.AddHostObjectToScript("bridge", new bridge());
            }
            catch (Exception ex)
            {

                Debug.Print($"Error in AddHostObjectToScript : {ex.Message}");
            }
        }
```
	
When I run the code, I catch no error in try catch block for AddHostObjectToScript.
So it seems to be good !?

Then I go to the DevTool page to see my HostedObject As mentionned in MS comments of AddHostObjectToScript :
 =>// Find added objects on the hostObjects property const bridge = chrome.webview.hostObjects.bridge;
So I get undefined in the DevTools console for:
 const bridge = chrome.webview.hostObjects.bridge;
If I run the command chrome.webview.hostObjects.length, I get 0 object found

Possible bug , or what I did wrong ?

Verdion of components used :
WebView2 control : 1.0.902.49
Edge Runtime is installed :
    ->Name : Microsoft Edge WebView2 Runtime
    ->Location : C:\Program Files (x86)\Microsoft\EdgeWebView\Application
    ->Version : 92.0.902.67
see also attached files :
![Screenshot_addHostObject](https://user-images.githubusercontent.com/86523070/128682105-94b9c923-b164-48f1-9997-245770639b09.png)
![Screenshot_DevToolsConsole](https://user-images.githubusercontent.com/86523070/128682118-05d4d6fb-9ed2-4765-8e7b-7060ebf1fd9f.png)

Thanks for your help

G.

**Comment by RickStrahl** [comment]
I think you need to create the instance and keep it alive in .NET. IOW, create a property on your host somewhere (form property or property on your handler class) and reference that rather than `new YourInteropObject()` as that reference goes out of scope.

**Comment by GrandStrateguerre** [comment]
Thanks Rick, I did what you suggested. Created a property to persist the hosted object. I also placed the declaration statement after the EnsureCoreWebView2Async await. See the code in the attachment.
But unfortunately I still can't get my hosted object to appear on the browser side.

I really don't see what can go wrong.

Maybe champnic of the team has an idea ?

G.

![New try suggested by rick](https://user-images.githubusercontent.com/86523070/129483113-b08510ae-a84c-41ed-9bbc-c960858a50cd.png)

**Comment by GrandStrateguerre** [comment]
Hi,

The result in the console made me confused :
![hostObjects length](https://user-images.githubusercontent.com/86523070/129541851-28d96f11-729d-4843-b9c7-ff4f4be4034c.png)
I expected after loading to find an object in the array chrome.webview.hostObjects so a .Length property different from 0 but it is not the case why ?

I saw this post https://github.com/MicrosoftEdge/WebView2Feedback/issues/494 which indicates that it works for WPF. So I took the same parameter and found that it works but with a curious behavior of the chrome.webview.hostObjects object seen above.

So I close this bug for AddHostObjectToScript;

However, I consider that there is a problem at the object level: chrome.webview.hostObjects because when we load an object into  chrome.webview.hostObjects the .length property always indicates 0 

I will open a new bug for chrome.webview.hostObjects

G.
